lib/hawkular/inventory/inventory_api_v4.rb

```diff
@@ -57,14 +57,26 @@ def children_resources(parent_id)
     # @return [Array<Resource>] List of resources
     def root_resources
       # FIXME: pagination => lazy-loaded list with ruby?
-      http_get('/resources?root=true')['results'].map { |r| Resource.new(r) }
+      resources root: "true"
+    end
+
+    # List resources
+    # @param [Hash] filter options to filter the resource list
+    # @option filter :root If truthy, only get root resources
+    # @option filter :feedId Filter by feed id
+    # @option filter :typeId Filter by type id
+    def resources(filter = {})
+      # FIXME: pagination => lazy-loaded list with ruby?
+      filter[:root] = !! filter[:root] if filter.key? :root
+      filter_query = '?' + filter.keys.join('=%s&') + '=%s' unless filter.empty?
+      http_get(url("/resources#{filter_query}", *filter.values))['results'].map { |r| Resource.new(r) }
     end
 
     # List resources for type
     # @return [Array<Resource>] List of resources
     def resources_for_type(type)
       # FIXME: pagination => lazy-loaded list with ruby?
-      http_get(url('/resources?typeId=', type))['results'].map { |r| Resource.new(r) }
+      resources typeId: type
```

Comment:
I don't like this, really. It makes for the params to be un-ruby. If we're going to use the params in any way, maybe they should be changed to be `type_id`, and be translated on the client layer.

Reply:
I was going to use https://apidock.com/rails/String/camelize but not sure if possible.

Reply:
Not yet, we had some talks on including ActiveSupport exactly for that, but I think for the time being we can just copy the method implementation.